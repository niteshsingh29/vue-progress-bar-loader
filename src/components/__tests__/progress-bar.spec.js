import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import ProgressBar from "./../.././components/progress-bar.vue";
import STATIC_VALUES from "../../enums/static.enum";

describe("Progress bar loader", () => {
  it("renders properly", () => {
    const wrapper = mount(ProgressBar, {
      props: {
        height: {
          type: String,
          default: STATIC_VALUES.HEIGHT,
        },
        width: {
          type: String,
          default: STATIC_VALUES.WIDTH,
        },
        backgroundColor: {
          type: String,
          default: STATIC_VALUES.BACKGROUNDCOLOR,
        },
        borderRadius: {
          type: String,
          default: STATIC_VALUES.BORDERADIUS,
        },
        loading: {
          type: Boolean,
          default: STATIC_VALUES.LOADING,
        },
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
