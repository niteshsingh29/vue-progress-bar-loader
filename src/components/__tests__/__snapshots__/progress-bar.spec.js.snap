// Vitest Snapshot v1

exports[`Progress bar loader > renders properly 1`] = `
<div
  class="outer"
  style="background-color: rgb(204, 204, 204); border-radius: [object Object];"
>
  <div
    class="inner"
    id="inner"
    style="width: 100%; border-radius: [object Object]; animation: progress 5000ms ease-in 1;"
  />
</div>
`;
